File: 2012/submissions/Khalil Fazal/1/Ruby/Main.rb
def firstIndex(word, input)
    if input.length > 0
        for $i in 0 .. input.length
            $j = 0

            while $j < word.length and $i + $j < input.length and input[$i + $j] == word[$j]
                $j += 1
            end

            if ($j == word.length)
                return $i
            end
        end
    end

    return -1
end


if ARGV.length > 0
    STDIN.each_line do |input|
        puts firstIndex(ARGV[0], input)
    end
end

File: 2012/submissions/Khalil Fazal/1/Perl/Main.pl
use feature qw(say);

sub firstIndex {
    my %args = @_;
    my $word = $args{word};
    my $input = $args{input};

    if (length($input) > 0) {
        for ($i = 0; $i < length($input); $i++) {
            $j = 0;

            while ($j < length($word) && $i + $j < length($input) && substr($input, $i + $j, 1) eq substr($word, $j, 1)) {
                $j++;
            }

            if ($j == length($word)) {
                return $i;
            }
        }
    }

    return -1;
}

sub main {
    if ($#ARGV > -1) {
        while(<STDIN>) {
            chomp;
            say(firstIndex(word => $ARGV[0], input => $_));
        }
    }
}

main();
